consent2/metadata.go

```diff
@@ -0,0 +1,129 @@
+package consent2
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/prebid/go-gdpr/consentconstants"
+)
+
+// Parse the metadata from the consent string.
+// This returns an error if the input is too short to answer questions about that data.
+func parseMetadata(data []byte) (consentMetadata, error) {
+	if len(data) < 30 {
```

Comment:
Nitpick: I think 29 bytes should be used as the minimum here. The highest bit needed to parse this section is 228.

Reply:
I think I made it 30 because there is a second bitfield we don't read yet, so practically it is not possible to have a 29 byte consent string if I counted right.